File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.swift
import Foundation

func solution(_ n:Int) -> Int
{
    let n_list = String(n, radix: 2).map{String($0)}
    var n_cnt = 0
    for c in n_list {
        if(c == "1") { n_cnt += 1 }
    }
    
    var answer = n+1
    while(true) {
        let a_list = String(answer, radix: 2).map{String($0)}
        var a_cnt = 0
        for c in a_list {
            if(c == "1") { a_cnt += 1 }
        }
        
        if(n_cnt == a_cnt) { break }
        else { answer += 1 }
    }
    
    return answer
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        char[] n_chars = Integer.toBinaryString(n).toCharArray();
        int n_cnt = 0;
        for(char c : n_chars) if(c == '1') n_cnt++;
        
        int answer = n+1;
        
        while(true) {
            char[] v = Integer.toBinaryString(answer).toCharArray();
            int v_cnt = 0;
            for(char c : v) if(c == '1') v_cnt++;
            
            if(n_cnt == v_cnt) break;
            else answer++;
        }
        return answer;
    }
}